src/modules/Api.ts

```diff
@@ -375,21 +411,31 @@ export class Api {
 
     const query: string = queryString.stringify(queryObj);
 
-    return request({
-      uri: `${this.endpoint}/v1/orders/${pair}?${query}`,
-      method: 'GET',
-      json: true,
+    const response = await axios({
+      url: `${this.endpoint}/v1/orders/${pair}?${query}`,
+      method: 'get',
       timeout: this.timeout,
     });
+
+    if (response.status !== 200) {
+      throw new Error(`API Error Status: ${response.status}`);
+    }
+
+    return response.data;
   }
 
   public async getMarkets(): Promise<{ markets: ApiMarket[] }> {
-    return request({
-      uri: `${this.endpoint}/v1/markets`,
-      method: 'GET',
-      json: true,
+    const response = await axios({
+      url: `${this.endpoint}/v1/markets`,
+      method: 'get',
       timeout: this.timeout,
     });
+
+    if (response.status !== 200) {
+      throw new Error(`API Error Status: ${response.status}`);
+    }
+
+    return response.data;
```

Comment:
Do you mind creating a function that wraps this behavior so we don't have to duplicate it everywhere?